### PR TITLE
Final Milestone : Space Travellers complete

### PR DESCRIPTION
**Pull request by Nemwel-Boniface**
## Hello Reviewer(s) :wave: :wave: 

This is @Nemwel-Boniface and @AckonSamuel team. In our group project we were able to accomplish the following:

The individual tasks are broken into two main blocks

## 1. Rockets 

- [x] Fetched data from the space-x fetching the [rockets](https://api.spacexdata.com/v3/rockets)
- [x] Displayed the rockets information including:
  - Rocket name
  - Rocket image
  - Rocket description

- [x] Included functionality to book rockets
- [x] Render components section conditionally letting users know when they have successfully booked a Rocket
- [x] Displayed the booked Rocket in the Myprofile page

## 2. Missions 

- [x]  Fetched data from the space-x fetching the [missions](https://api.spacexdata.com/v3/missions)
- [x] Displayed the Mission information including:
  - Mission name
  - Mission description

- [x] Included functionality to join Missions
- [x] Render components section conditionally letting users know when they have successfully joined a Mission
- [x] Displayed the booked Missions on the My profile page

Besides this, we also wrote tests for our project testing:

- [x] The Rockets components
- [x] The Missions components
- [x] The My profile component

## 3. Other requirements 

- [x] We also did documentation of our project by updating our README.md file to reflect all the information that a user needs about the project.

- [x] We also deployed the project for testing purposes

- [x] We made sure to meet all the project requirements

- [x] We made sure to check and fix all linter errors

Thank you for taking the time to review our project :pray:

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
+import React, { useEffect } from 'react';
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
+import { Routes, Route } from 'react-router-dom';
+import Rockets from './components/Rockets/rockets';
+import MyProfile from './components/myprofile/Myprofile';
+import Header from './components/header/Header';
+import { getRocketFromAPI } from './redux/Rockets/rockets';
+import Appmissions from './Appmissions';
 
 function App() {
```
Comment (Deepakdanger) [inline src/App.js]: - [x] Kindly use ES6.

File: src/Appjus.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React, { useEffect } from 'react';
+import { shallowEqual, useDispatch, useSelector } from 'react-redux';
+import Rockets from './components/Rockets/rockets';
+import { getRocketFromAPI } from './redux/Rockets/rockets';
+
+function Appjus() {
```
Comment (Deepakdanger) [inline src/Appjus.js]: - [x] Kindly use ES6.

File: src/Appbus.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import MyProfile from './components/myprofile/Myprofile';
+
+function Appprofile() {
```
Comment (Deepakdanger) [inline src/Appbus.js]: - [x] Kindly use ES6.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
**Hi @Team**, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Good PR :+1: 

## Required changes
- [x] After opening the website, I do one reserve rocket (i.e Falcon 1), After moving to `My profile` `Falcon 1` is present, But when I move back to `Rockets`, the rocket list vanishes, only reserve Rockets is present there, Please correct it.

- [x] _Check the comments under the review._


## Optional changes
You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Hamzaoutdoors — APPROVED** [review]
Hi team, 

## Congratulations! 🎉 Project Approved ✅

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚀

To Highlight:

- [x] The Pull Request has a proper title and description. ✔️
- [x] You have added a descriptive Readme file. Good Job. ✔️
- [x] There is no linter error present. ✔️
- [x] Project has met all the technical aspects you needed for this project! ✔️
- [x] Great job implementing redux and adding unit tests to your code ✔️  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] I think it will be better if you can provide a link to your live demo to this project, let the world explore it 😄 


_Cheers and Happy coding and Keep rocking!_ 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Nemwel-Boniface** [comment]
Thank you @Hamzaoutdoors  for the review. 
I will update it on the main branch :+1:

**Pull request merged**